src/ray/raylet/scheduling/cluster_task_manager.h

```diff
@@ -160,7 +171,7 @@ class ClusterTaskManager : public ClusterTaskManagerInterface {
   absl::flat_hash_map<SchedulingClass, std::deque<std::shared_ptr<internal::Work>>>
       infeasible_tasks_;
 
-  const SchedulerResourceReporter scheduler_resource_reporter_;
+  std::unique_ptr<SchedulerResourceReporter> scheduler_resource_reporter_;
```

Comment:
I think it should be `std::shared_ptr` here, otherwise the `std::shared_ptr<ClusterResourceScheduler> GetClusterResourceScheduler() const` will got a compile error.

Reply:
This is scheduler_resource_**reporter**, not scheduler_resource_**scheduler**.